lib/services/database/comment_repository_service.dart

```diff
@@ -0,0 +1,124 @@
+import "package:proxima/models/database/post/post_id_firestore.dart";
+import "package:proxima/services/database/firestore_service.dart";
+
+/// This class is a service that allows to interact with the comments
+/// of the posts in the firestore database.
+class CommentRepositoryService {
+  final FirebaseFirestore _firestore;
+
+  CommentRepositoryService({required FirebaseFirestore firestore})
+      : _firestore = firestore;
+
+  /// Returns the document reference of the post with id [postId]
+  DocumentReference<Map<String, dynamic>> _postDocument(
+    PostIdFirestore postId,
+  ) {
+    return _firestore
+        .collection(PostFirestore.collectionName)
+        .doc(postId.value);
+  }
+
+  /// Returns the collection reference of the subcollection of comments of
+  /// the post with id [parentPostId]
+  CollectionReference<Map<String, dynamic>> _commentsSubCollection(
+    PostIdFirestore parentPostId,
+  ) {
+    return _postDocument(parentPostId)
+        .collection(CommentFirestore.subCollectionName);
+  }
+
+  /// This method returns the comments of the post with id [parentPostId]
+  Future<List<CommentFirestore>> getComments(
+    PostIdFirestore parentPostId,
+  ) async {
+    final commentsQuery = await _commentsSubCollection(parentPostId).get();
+
+    final comments = commentsQuery.docs
+        .map((docSnap) => CommentFirestore.fromDb(docSnap))
+        .toList();
+
+    return comments;
+  }
+
+  /// This method will add the comment with data [commentData] to the
+  /// post with id [parentPostId].
+  /// It will also update the number of comments of the post.
+  /// This is done atomically.
+  ///
+  /// The method returns the id of the comment that was added.
+  Future<CommentIdFirestore> addComment(
+    PostIdFirestore parentPostId,
+    CommentData commentData,
+  ) async {
+    // Generate a new reference for the comment
+    // Although generated locally, the new id can be considered unique
+    // https://stackoverflow.com/questions/54268257/what-are-the-chances-for-firestore-to-generate-two-identical-random-keys
+    final newCommentRef = _commentsSubCollection(parentPostId).doc();
+
+    // Create a batch write to perform the operations atomically
+    final batch = _firestore.batch();
+
+    batch.set(newCommentRef, commentData.toDbData());
+
+    final postDocRef = _postDocument(parentPostId);
+    batch.update(
+      postDocRef,
+      {PostData.commentCountField: FieldValue.increment(1)},
+    );
+
+    await batch.commit();
+
+    return CommentIdFirestore(value: newCommentRef.id);
+  }
+
+  /// This method will delete the comment with id [commentId] from the
+  /// post with id [parentPostId].
+  /// It will also update the number of comments of the post.
+  /// This is done atomically.
+  /// If the comment does not exist, the method will do nothing.
+  Future<void> deleteComment(
+    PostIdFirestore parentPostId,
+    CommentIdFirestore commentId,
+  ) async {
```

Comment:
Here we can remote the `async` clause of the function and simply return the same `Future` that `_firestore.runTransaction` emits. (like `return _firestore.runTransaction`)

Reply:
You are right, actually I think we can even use the arrow notiation here.
Thus I use it in 2d01850c7f0221d6121ec633d3c6587df1546db6.